more_itertools/more.py

```diff
@@ -1898,3 +1901,68 @@ def inner_wrapper(*args, **kwargs):
         return outer_wrapper
 
     return decorator
+
+
+class classify(object):
+    """Wrap *iterable* and return an object that progressively saves its items
```

Comment:
We could say "return a mapping" to be a bit more specific, or "a mapping-like object" if we don't fulfill the whole contract.